### ISSUE TITLE
"ungrouped" always contains all hosts even though they belong to another group.

### ISSUE THREAD

**Issue by iangregsondev** [relex/aini#2]
Hi,

great library!

For my use case I don't need "all" and "ungrouped" - so its not a major thing for me  BUT...

I am iterating over the groups and outputting all group names and their respective hosts, I noticed that ungrouped is output with all hosts. According to the docs at ansible, ungrouped should only contain hosts that don't belong to any other group.

Here is a short sample of my quick and dirty method for demonstrating..

```
	for _, g := range inventory.Groups {
		_, _ = datawriter.WriteString("[" + g.Name + "]\n")
		for _, h := range g.Hosts {
			_, _ = datawriter.WriteString(h.Name + "\n")

.....
```

So basically I get a list of groups and their respective hosts, works great! But I also get an "ungrouped" group but it contains ALL hosts

Is this correct? I would have suspected that it should be empty as all my hosts belong to another group.

Thanks in advance

**Comment by bemyak** [comment]
Hi, @iangregsondev ! Thanks for your report.

I [just added](https://github.com/relex/aini/commit/d0235fcb71379304f05e2b33cb2061e06a2ef57d) a few tests for this case and all of them are passing, so I can't really reproduce this. Can you please provide a sample of you inventory file that you're parsing?

**Comment by iangregsondev** [comment]
Hi, sure, its called "test.ini", my inventory file is quite big but I did a simple test with this and the same happens.

```
swarm-1	ansible_user=root ansible_host=888.888.888.88

[something_node_pruned]
swarm-1
```


so basically what I am doing is writing the file in another place and this is what is produced.

```
; name of file
swarm-1 ansible_host=888.888.888.88 ansible_user=root

[ungrouped]
swarm-1

[something_node_pruned]
swarm-1

[all]
swarm-1

```

I of course would be ignoring the "all" and "ungrouped" but this is just a test.

As you can see the "ungrouped" is there but the host is part of another group.

**Comment by bemyak** [comment]
Thanks for the details, this should be fixed now in `v1.2.1`
